Add stat archetype and modslot to item tile

bhollis opened the pull request:
<!-- To add entries to the CHANGELOG.md, include lines in your commit messages that start with `Changelog:` followed by the description -->

This is pretty rough but I want opinions on how/whether to show this info on item tiles:

<img width="171" height="143" alt="Screenshot 2025-09-18 at 10 59 52 PM" src="https://github.com/user-attachments/assets/a5f7962d-d29b-400e-b75a-288d008bf28b" />

bhollis commented:
<img width="171" height="215" alt="Screenshot 2025-09-19 at 11 11 04 PM" src="https://github.com/user-attachments/assets/68eb53aa-d237-4239-a5ab-9e94d3be955b" />

OK, yeah, I like that better

chainrez commented:
AS OF NOW, these things are exclusive of each other and *could* occupy the same corner (artifice/archetype/activity mod).

This idea begs the question of whether set bonuses could be displayed, as they are basically the new armor "activity mod" type thing. But the logistics are very messy because sets don't have icons, *perks* have icons (2 per set), and honestly they are not all very visually distinct/self explanatory.

The small icons, transparency, and and size, can interact negatively with icons that have fine detail in the same area. These icons might need a dark background corner behind them.

I'm not fully sold on including archetype on icons, obligatory complaining about busyness, but it _is_ the obvious thing to add to them, and some people will probably really like it. Such a small set of variations (6 archetypes) would make a great inventory sort, and it would naturally sort old style armor from new, if you included that corner as part of the sort.

bhollis commented:
How's this? Moved them both to the top right and added a stronger shadow.

<img width="232" height="219" alt="Screenshot 2025-09-27 at 8 35 09 PM" src="https://github.com/user-attachments/assets/4e07c2ee-bc39-4c48-9d91-bfec6f7f0822" />

A black square behind them also works but is slightly more aggressive:
<img width="231" height="221" alt="Screenshot 2025-09-27 at 8 46 42 PM" src="https://github.com/user-attachments/assets/a30772e9-bd44-4319-9b49-da3df4a91986" />

The other obvious place would be in the polaroid, but I know you wanted to reserve that for thumbs:
<img width="233" height="224" alt="Screenshot 2025-09-27 at 8 49 13 PM" src="https://github.com/user-attachments/assets/ba512a9d-223a-41e9-a7ae-ca6c71d2ed3e" />

bhollis commented:
Do you have a preference among the four alternatives I showed?

bhollis commented:
OK. I'm going to merge this now for feedback and I'll iterate on contrast options.
